atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraKeyValueServiceImpl.java

```diff
@@ -257,8 +257,10 @@ protected CassandraKeyValueServiceImpl(Logger log,
                                        Optional<CassandraJmxCompactionManager> compactionManager,
                                        Optional<LeaderConfig> leaderConfig,
                                        boolean initializeAsync) {
-        super(AbstractKeyValueService.createFixedThreadPool("Atlas Cassandra KVS",
-                configManager.getConfig().poolSize() * configManager.getConfig().servers().size()));
+        super(AbstractKeyValueService.createScalingThreadPoolWithCallerRunsOnEmpty(
+                "Atlas Cassandra KVS",
+                configManager.getConfig().poolSize() * configManager.getConfig().servers().size(),
+                configManager.getConfig().maxConnectionBurstSize() * configManager.getConfig().servers().size()));
```

Comment:
I'm thinking it might be beneficial to isolate the executor to C*KVS for now, e.g.:

```
    private static ExecutorService createExecutor(CassandraKeyValueServiceConfigManager configManager) {
        return newScalingThreadPoolWithCallerRunsOnEmpty(
                new NamedThreadFactory("Atlas Cassandra KVS", false),
                configManager.getConfig().poolSize() * configManager.getConfig().servers().size(),
                configManager.getConfig().maxConnectionBurstSize() * configManager.getConfig().servers().size());
    }

    /**
     * Creates a thread pool that maintains between corePoolSize and maximumPoolSize threads. When maximumPoolSize is
     * hit, new requests will not be queued but rather will be run in the caller's thread.
     *
     * @return the newly created thread pool
     */
    private static ThreadPoolExecutor newScalingThreadPoolWithCallerRunsOnEmpty(
            ThreadFactory threadFactory,
            int corePoolSize, int maximumPoolSize) {
        return PTExecutors.newThreadPoolExecutor(
                corePoolSize,
                maximumPoolSize,
                1,
                TimeUnit.MINUTES,
                new ArrayBlockingQueue<Runnable>(0),
                threadFactory,
                new ThreadPoolExecutor.CallerRunsPolicy());
    }

    protected CassandraKeyValueServiceImpl(Logger log,
                                       CassandraKeyValueServiceConfigManager configManager,
                                       Optional<CassandraJmxCompactionManager> compactionManager,
                                       Optional<LeaderConfig> leaderConfig,
                                       boolean initializeAsync) {
        super(createExecutor(configManager));
...
```